canopeum_frontend/src/components/analytics/BatchTable.tsx

```diff
@@ -106,8 +106,8 @@ const BatchTable = (props: Props) => {
                 style={{ borderColor: cellBorderColor }}
               >
                 <ul>
-                  {batch.species.map(type => (
-                    <li key={`batch-${batch.id}-treeType-${type.en}`}>
+                  {batch.species.map((type, i) => (
+                    <li key={`batch-${batch.id}-treeType-${type.en}-${i}`}>
```

Comment:
Is there some other unique identifier to the species we can use here? The point is to help React know what and when to re-render, if you swap the order of two items that had the same ID but differentiated on their index, then they'll end up with the same key and not swap visually in the UI because React will think it doesn't need to re-render !
https://react.dev/learn/rendering-lists#where-to-get-your-key
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md

Reply:
You are true , it is not event necessary because a batch shouldn't have same treetype twice (was the case because of development in progress of edit batch)